protocol/syncer_test.go

```diff
@@ -287,26 +287,23 @@ func TestWatchSyncWithPeer(t *testing.T) {
 			assert.NotNil(t, peer)
 
 			latestBlock := newBlocks[len(newBlocks)-1]
-			doneCh := make(chan struct{}, 1)
-			go func() {
-				syncer.WatchSyncWithPeer(peer, func(b *types.Block) bool {
-					// sync until latest block
-					return b.Header.Number >= latestBlock.Header.Number
-				})
-				// wait until syncer updates status by latest block
-				WaitUntilProcessedAllEvents(t, syncer, 10*time.Second)
-				doneCh <- struct{}{}
-			}()
-
-			select {
-			case <-doneCh:
+			startSyncTime := time.Now()
+			endSyncTime := startSyncTime.Add(time.Second * 5)
+			syncer.WatchSyncWithPeer(peer, func(b *types.Block) bool {
+				if time.Now().After(endSyncTime) {
+					// Timeout
+					return true
+				}
+				// sync until latest block
+				return b.Header.Number >= latestBlock.Header.Number
+			})
+
+			if tt.synced {
```

Comment:
This is inherently wrong, since tt.synced is something that is set in the test specification struct. Please fix the assertion logic

Reply:
You're right 👍 

Fixed the asserts in the original test:

ddc047b